nimbleparse/src/main.rs

```diff
@@ -141,12 +141,12 @@ fn main() {
     };
     let entry = match header.entry("yacckind".to_string()) {
         Entry::Occupied(_) => unreachable!("Header should be empty"),
-        Entry::Vacant(v) => v.occupied_entry(),
+        Entry::Vacant(v) => v,
```

Comment:
I think rather than what I discussed in the commit message, of adding a different `key_entry`, it's best to just implement the `mark_required` and `set_merge_behavior` functions for `VacantEntry` directly, perhaps `mark_used` too but I can't think of a reason to mark a vacant entry used.

I still think it's best to do as a separate patch, but let me know if you feel otherwise.

Reply:
Doing it in another PR is fine!